migration-engine/core/src/lib.rs

```diff
@@ -105,7 +106,10 @@ pub async fn qe_setup(prisma_schema: &str) -> CoreResult<()> {
         .first()
         .ok_or_else(|| CommandError::Generic(anyhow::anyhow!("There is no datasource in the schema.")))?;
 
-    match &source.active_provider {
+    // 1. creates schema & database
+    SqlMigrationConnector::qe_setup(&source.url().value).await?;
```

Comment:
Nitpick: moving this before line 122, inside the match, would avoid using the SqlMigrationConnector in a context that should be database agnostic.